index.js

```diff
@@ -1,14 +1,33 @@
 'use strict';
 
-const createDelay = willResolve => (ms, value) => {
+const createAbortError = () => {
+	const error = new Error('Delay aborted');
+	error.name = 'AbortError';
+	return error;
+};
+
+const createDelay = willResolve => (ms, {value, signal} = {}) => {
+	if (signal && signal.aborted) {
+		return Promise.reject(createAbortError());
+	}
+
 	let timeoutId;
 	let settle;
+	let rejectFn;
 
 	const delayPromise = new Promise((resolve, reject) => {
 		settle = willResolve ? resolve : reject;
+		rejectFn = reject;
 		timeoutId = setTimeout(settle, ms, value);
 	});
 
+	if (signal) {
+		signal.addEventListener('abort', () => {
+			clearTimeout(timeoutId);
+			rejectFn(createAbortError());
+		});
```

Comment:
Should we use `});` => `}, {once: true});` here so it unsubscribes after aborting?

Reply:
Hmm, yes I guess

Reply:
Done